Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -22,6 +28,171 @@ def __init__(self, port):
+        response: Response from the SDK.
+        protocol: Communication protocol.
+
+    Returns:
+        markdown string to be displayed in the war room.
+    """
+    hr_output = []
+    for traffic in response:
+        hr_output.append({
+            "Source IP": traffic.get('src', {}).get('ip'),
+            "Destination IP": traffic.get('dst', {}).get('ip'),
+            "Destination Workload Hostname": traffic.get('dst', {}).get('workload', {}).get('hostname'),
+            "Service Port": traffic.get('service', {}).get('port'),
+            "Service Protocol": protocol,
+            "Policy Decision": traffic.get('policy_decision'),
+            "State": traffic.get("state"),
+            "Flow Direction": traffic.get('flow_direction'),
+            "First Detected": arg_to_datetime(traffic.get('timestamp_range', {}).get('first_detected', '')).strftime(  # type: ignore # noqa
+                HR_DATE_FORMAT),  # type: ignore # noqa
+            "Last Detected": arg_to_datetime(traffic.get('timestamp_range', {}).get('last_detected', '')).strftime(  # type: ignore # noqa
+                HR_DATE_FORMAT)
+        })
+
+    headers = ["Source IP", "Destination IP", "Destination Workload Hostname", "Service Port", "Service Protocol",
+               "Policy Decision", "State", "Flow Direction", "First Detected", "Last Detected"]
+
+    return tableToMarkdown("Traffic Analysis:", hr_output, headers=headers, removeNull=True)
+
+
+def prepare_hr_for_illumio_virtual_service_create_command(result: dict) -> str:
+    """Prepare human-readable output for illumio virtual service create command.
+
+    Args:
+        result: result returned after creating Virtual Service.
+
+    Returns:
+        human-readable output.
+    """
+    table = {
+        num: name[len(SOCKET_PREFIX):]
+        for name, num in vars(socket).items()
+        if name.startswith(SOCKET_PREFIX)
+    }
+    hr_outputs = []
+    for service_port in result.get("service_ports", []):
+        hr_outputs.append({
+            "Virtual Service HREF": result.get("href"),
+            "Created At": arg_to_datetime(result.get("created_at"), '').strftime(HR_DATE_FORMAT),  # type: ignore
+            "Updated At": arg_to_datetime(result.get("updated_at", '')).strftime(HR_DATE_FORMAT),  # type: ignore
+            "Name": result.get("name"),
+            "Description": result.get("description"),
+            "Service Port": service_port.get("port")
+            if "port" in service_port
+            else "all ports have been selected",
+            "Service Protocol": table.get(service_port.get("proto"))})
+    headers = ["Virtual Service HREF", "Created At", "Updated At", "Name", "Description", "Service Port",
+               "Service Protocol"]
+    title = f'Virtual Service:\n#### Successfully created virtual service: {result.get("href")}\n'
+    return tableToMarkdown(title, hr_outputs, headers=headers, removeNull=True)
+
+
+def validate_required_parameters(**kwargs) -> None:
```

Comment:
Are required parameters not validated on the front-end?

Reply:
No, if you pass an empty string like `" "` it'll not validate. hence we need this function.

Reply:
Ok, makes sense.